contracts/Property.sol

```diff
@@ -138,10 +138,15 @@ contract Property is
         );
 
         // Update a new booking record
+        uint256 referralFeeNumerator = management.referralFeeNumerator();
         BookingInfo storage bookingInfo = booking[_setting.bookingId];
         bookingInfo.checkIn = _setting.checkIn;
         bookingInfo.checkOut = _setting.checkOut;
         bookingInfo.balance = _setting.bookingAmount;
+        bookingInfo.feeNumerator = management.feeNumerator();
+        if (referralFeeNumerator != 0) {
+            bookingInfo.referralFeeNumerator = referralFeeNumerator;
```

Comment:
There's no need to cache referral fee into `referralFeeNumerator` and validate it. Just update `bookingInfo.referralFeeNumerator` if referrer exists below.